File: src/lib/bridge/theme.ts
import { invoke } from "@tauri-apps/api/core";
import type { CustomTheme } from "@wenyan-md/ui";

export async function loadThemes(): Promise<CustomTheme[]> {
    return await invoke("load_themes");
}

export async function saveTheme(id: string | null, name: string, css: string): Promise<string> {
    return await invoke("save_theme", { id, name, css });
}

export async function removeTheme(id: string): Promise<void> {
    await invoke("remove_theme", { id });
}

File: src-tauri/src/commands/theme.rs
use crate::dto::theme::ThemeDto;
use crate::error::AppResult;
use crate::infrastructure::db::DbManager;
use crate::infrastructure::repositories::theme::ThemeRepository;
use tauri::State;

#[tauri::command]
pub async fn load_themes(db: State<'_, DbManager>) -> AppResult<Vec<ThemeDto>> {
    let repo = ThemeRepository::new(&db);
    repo.load_all().await
}

#[tauri::command]
pub async fn save_theme(
    db: State<'_, DbManager>,
    id: Option<String>,
    name: String,
    css: String,
) -> AppResult<String> {
    let repo = ThemeRepository::new(&db);
    repo.save(id, &name, &css).await
}

#[tauri::command]
pub async fn remove_theme(db: State<'_, DbManager>, id: String) -> AppResult<()> {
    let repo = ThemeRepository::new(&db);
    repo.remove(id).await
}
